bigquery/google/cloud/bigquery/client.py

```diff
@@ -1929,6 +1935,60 @@ def list_rows(
         )
         return row_iterator
 
+    def schema_from_json(self, file_or_path):
+        """Takes a file object or file path that contains json that describes
+            a table schema.
+
+            Returns:
+                List of schema field objects.
+        """
+        file_obj = None
+        json_data = None
+        schema_field_list = list()
+
+        if isinstance(file_or_path, io.IOBase):
+            file_obj = file_or_path
+        else:
+            try:
+                file_obj = open(file_or_path)
+            except OSError:
+                raise TypeError(_NEED_JSON_FILE_ARGUMENT)
+
+        try:
+            json_data = json.load(file_obj)
+        except JSONDecodeError:
+            raise TypeError(_NEED_JSON_FILE_ARGUMENT)
```

Comment:
I'd prefer if we let this raise (don't catch it). That way the user doesn't lose context about what is wrong with the input file.

Reply:
Took this piece out.